app/models/accounting/quickbooks/consumer.rb

```diff
@@ -6,33 +6,23 @@ module Quickbooks
     class Consumer
       def initialize
         if ::Quickbooks.sandbox_mode
-          oauth_consumer_key = ENV.fetch('QB_SANDBOX_OAUTH_CONSUMER_KEY')
-          oauth_consumer_secret = ENV.fetch('QB_SANDBOX_OAUTH_CONSUMER_SECRET')
+          oauth_consumer_key = ENV.fetch('QB_SANDBOX_OAUTH2_CLIENT_ID')
+          oauth_consumer_secret = ENV.fetch('QB_SANDBOX_OAUTH2_CLIENT_SECRET')
         else
-          oauth_consumer_key = ENV.fetch('QB_OAUTH_CONSUMER_KEY')
-          oauth_consumer_secret = ENV.fetch('QB_OAUTH_CONSUMER_SECRET')
+          oauth_consumer_key = ENV.fetch('QB_OAUTH2_CLIENT_ID')
+          oauth_consumer_secret = ENV.fetch('QB_OAUTH2_CLIENT_SECRET')
         end
 
-        @oauth_consumer = OAuth::Consumer.new(oauth_consumer_key, oauth_consumer_secret,
-          site: 'https://oauth.intuit.com',
-          request_token_path: '/oauth/v1/get_request_token',
-          authorize_url: 'https://appcenter.intuit.com/Connect/Begin',
-          access_token_path: '/oauth/v1/get_access_token'
-        )
+        @oauth_consumer = OAuth2::Client.new(oauth_consumer_key, oauth_consumer_secret,
+          site: "https://appcenter.intuit.com/connect/oauth2",
+          authorize_url: "https://appcenter.intuit.com/connect/oauth2",
+          token_url: "https://oauth.platform.intuit.com/oauth2/v1/tokens/bearer")
       end
```

Comment:
In general I try to steer clear of hardcoding urls like this, but I'm assuming there's a good reason to do it in this case.